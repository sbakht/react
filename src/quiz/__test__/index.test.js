import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Quiz from '../index';
import toJSON from 'enzyme-to-json';

const questions = [
  {
    text: "Who is an adc?",
    choices: [
      { val: "a", text: "riven", isWrong: true },
      { val: "b", text: "zed", isWrong: true },
      { val: "c", text: "kogmaw", isCorrect: true },
      { val: "d", text: "janna", isWrong: true }
    ]
  },
  {
    text: "Who is a mid?",
    choices: [
      { val: "a", text: "riven", isWrong: true },
      { val: "b", text: "zed", isCorrect: true },
      { val: "c", text: "kogmaw", isWrong: true },
      { val: "d", text: "janna", isWrong: true }
    ]
  }
];

test('render question', () => {
  var component = shallow(<Quiz questions={questions}  />);
  expect(toJSON(component)).toMatchSnapshot();
});

test('render results', () => {
  var component = shallow(<Quiz questions={questions}  />);
  component.instance().showResults();
  component.update();
  expect(toJSON(component)).toMatchSnapshot();
});

test('change question', () => {
  var component = shallow(<Quiz questions={questions}  />);
  component.instance().nextQuestion();
  component.update();
  expect(toJSON(component)).toMatchSnapshot();

  component.instance().prevQuestion();
  component.update();
  expect(toJSON(component)).toMatchSnapshot();
});

test('update answers', () => {
  var component = shallow(<Quiz questions={questions}  />);
  component.instance().onAnswer(questions[0].choices[1])
  component.update();
  expect(toJSON(component)).toMatchSnapshot();
});

test('change question -> update answers', () => {
  var component = shallow(<Quiz questions={questions}  />);
  component.instance().nextQuestion();
  component.update();
  expect(toJSON(component)).toMatchSnapshot();

  component.instance().onAnswer(questions[1].choices[0])
  component.update();
  expect(toJSON(component)).toMatchSnapshot();
});

test('has proptypes', () => {
  expect(Quiz.propTypes).toBeDefined();
  
  Object.keys(Quiz.propTypes).map((key) => {
    expect(Quiz.propTypes[key]).toBeDefined();
  });
});