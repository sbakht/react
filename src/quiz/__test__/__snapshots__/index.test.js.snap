// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change question -> update answers 1`] = `
<div
  className="quiz"
>
  <div>
    <div>
      <p>
        Question 
        2
         of 
        2
      </p>
      <QuizQuestion
        allowDoOver={true}
        answer={Object {}}
        onAnswer={[Function]}
        question={
          Object {
            "choices": Array [
              Object {
                "isWrong": true,
                "text": "riven",
                "val": "a",
              },
              Object {
                "isCorrect": true,
                "text": "zed",
                "val": "b",
              },
              Object {
                "isWrong": true,
                "text": "kogmaw",
                "val": "c",
              },
              Object {
                "isWrong": true,
                "text": "janna",
                "val": "d",
              },
            ],
            "text": "Who is a mid?",
          }
        }
      />
    </div>
    <div>
      <Pager
        onNext={[Function]}
        onPrev={[Function]}
        qLen={2}
        qNum={2}
        showResults={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`change question -> update answers 2`] = `
<div
  className="quiz"
>
  <div>
    <div>
      <p>
        Question 
        2
         of 
        2
      </p>
      <QuizQuestion
        allowDoOver={true}
        answer={
          Object {
            "isWrong": true,
            "text": "riven",
            "val": "a",
          }
        }
        onAnswer={[Function]}
        question={
          Object {
            "choices": Array [
              Object {
                "isWrong": true,
                "text": "riven",
                "val": "a",
              },
              Object {
                "isCorrect": true,
                "text": "zed",
                "val": "b",
              },
              Object {
                "isWrong": true,
                "text": "kogmaw",
                "val": "c",
              },
              Object {
                "isWrong": true,
                "text": "janna",
                "val": "d",
              },
            ],
            "text": "Who is a mid?",
          }
        }
      />
    </div>
    <div>
      <Pager
        onNext={[Function]}
        onPrev={[Function]}
        qLen={2}
        qNum={2}
        showResults={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`change question 1`] = `
<div
  className="quiz"
>
  <div>
    <div>
      <p>
        Question 
        2
         of 
        2
      </p>
      <QuizQuestion
        allowDoOver={true}
        answer={Object {}}
        onAnswer={[Function]}
        question={
          Object {
            "choices": Array [
              Object {
                "isWrong": true,
                "text": "riven",
                "val": "a",
              },
              Object {
                "isCorrect": true,
                "text": "zed",
                "val": "b",
              },
              Object {
                "isWrong": true,
                "text": "kogmaw",
                "val": "c",
              },
              Object {
                "isWrong": true,
                "text": "janna",
                "val": "d",
              },
            ],
            "text": "Who is a mid?",
          }
        }
      />
    </div>
    <div>
      <Pager
        onNext={[Function]}
        onPrev={[Function]}
        qLen={2}
        qNum={2}
        showResults={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`change question 2`] = `
<div
  className="quiz"
>
  <div>
    <div>
      <p>
        Question 
        1
         of 
        2
      </p>
      <QuizQuestion
        allowDoOver={true}
        answer={Object {}}
        onAnswer={[Function]}
        question={
          Object {
            "choices": Array [
              Object {
                "isWrong": true,
                "text": "riven",
                "val": "a",
              },
              Object {
                "isWrong": true,
                "text": "zed",
                "val": "b",
              },
              Object {
                "isCorrect": true,
                "text": "kogmaw",
                "val": "c",
              },
              Object {
                "isWrong": true,
                "text": "janna",
                "val": "d",
              },
            ],
            "text": "Who is an adc?",
          }
        }
      />
    </div>
    <div>
      <Pager
        onNext={[Function]}
        onPrev={[Function]}
        qLen={2}
        qNum={1}
        showResults={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`render question 1`] = `
<div
  className="quiz"
>
  <div>
    <div>
      <p>
        Question 
        1
         of 
        2
      </p>
      <QuizQuestion
        allowDoOver={true}
        answer={Object {}}
        onAnswer={[Function]}
        question={
          Object {
            "choices": Array [
              Object {
                "isWrong": true,
                "text": "riven",
                "val": "a",
              },
              Object {
                "isWrong": true,
                "text": "zed",
                "val": "b",
              },
              Object {
                "isCorrect": true,
                "text": "kogmaw",
                "val": "c",
              },
              Object {
                "isWrong": true,
                "text": "janna",
                "val": "d",
              },
            ],
            "text": "Who is an adc?",
          }
        }
      />
    </div>
    <div>
      <Pager
        onNext={[Function]}
        onPrev={[Function]}
        qLen={2}
        qNum={1}
        showResults={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`render results 1`] = `
<div
  className="quiz"
>
  <Results
    answers={Array []}
    questions={
      Array [
        Object {
          "choices": Array [
            Object {
              "isWrong": true,
              "text": "riven",
              "val": "a",
            },
            Object {
              "isWrong": true,
              "text": "zed",
              "val": "b",
            },
            Object {
              "isCorrect": true,
              "text": "kogmaw",
              "val": "c",
            },
            Object {
              "isWrong": true,
              "text": "janna",
              "val": "d",
            },
          ],
          "text": "Who is an adc?",
        },
        Object {
          "choices": Array [
            Object {
              "isWrong": true,
              "text": "riven",
              "val": "a",
            },
            Object {
              "isCorrect": true,
              "text": "zed",
              "val": "b",
            },
            Object {
              "isWrong": true,
              "text": "kogmaw",
              "val": "c",
            },
            Object {
              "isWrong": true,
              "text": "janna",
              "val": "d",
            },
          ],
          "text": "Who is a mid?",
        },
      ]
    }
  />
</div>
`;

exports[`update answers 1`] = `
<div
  className="quiz"
>
  <div>
    <div>
      <p>
        Question 
        1
         of 
        2
      </p>
      <QuizQuestion
        allowDoOver={true}
        answer={
          Object {
            "isWrong": true,
            "text": "zed",
            "val": "b",
          }
        }
        onAnswer={[Function]}
        question={
          Object {
            "choices": Array [
              Object {
                "isWrong": true,
                "text": "riven",
                "val": "a",
              },
              Object {
                "isWrong": true,
                "text": "zed",
                "val": "b",
              },
              Object {
                "isCorrect": true,
                "text": "kogmaw",
                "val": "c",
              },
              Object {
                "isWrong": true,
                "text": "janna",
                "val": "d",
              },
            ],
            "text": "Who is an adc?",
          }
        }
      />
    </div>
    <div>
      <Pager
        onNext={[Function]}
        onPrev={[Function]}
        qLen={2}
        qNum={1}
        showResults={[Function]}
      />
    </div>
  </div>
</div>
`;
